### PR TITLE
Add support for an app_id to stabely identify applications.

### PR DESCRIPTION
**Pull request by TimothyBJacobs**
Trac ticket: https://core.trac.wordpress.org/ticket/51583

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/authorize-application.php [hunk 1]
```diff
@@ -18,6 +18,7 @@
 	$success_url = $_POST['success_url'];
 	$reject_url  = $_POST['reject_url'];
 	$app_name    = $_POST['app_name'];
+	$app_id      = ! empty( $_REQUEST['app_id'] ) ? $_REQUEST['app_id'] : '';
```
Comment (georgestephanis) [inline src/wp-admin/authorize-application.php:21]: Should this be `$_POST` ?  Is there any reason we'd accept from either COOKIE or GET or the like?

Comment (georgestephanis) [reply src/wp-admin/authorize-application.php:21]: Also, as this is handling the form we build down below, is there any circumstance where it wouldn't be set?  I think we can just do `$app_id      = $_POST['app_id'];` instead -- possibly with a `(int)`

Comment (TimothyBJacobs) [reply src/wp-admin/authorize-application.php:21]: Good point, we can simplify to `$_POST['app_id']`.

**Comment by TimothyBJacobs** [comment]
Fixed in fe2053f2c1cff0c416112103988e832687ca3836.

**Pull request closed without merging**